### DISCUSSION TITLE
color scheme: function color

### DISCUSSION THREAD

**Discussion by peterjosvai** [geany/geany#3840] [Q&A]
hi, 

I see this block:   (in a theme):

type=#2C8E99;;true;false
class=type
function=#FFFF00
parameter=#FFF35F

and can even guess that the second "true" stands for "bold"...
But I can't get the functions' color change  (I mean my functions, not php's)

Is there a way to do that?

thanks in advance...

Peter

// I've done a search in Q&A only....

**Comment by elextr** [comment]
Not knowing what language you are using its hard to help, does the lexer recognise style something as "function" ?

**Comment by peterjosvai** [comment]
hi,
  
sorry, I'm talking about php/html...
  

  

  

  
  
  
  
>

**Comment by elextr** [comment]
The Php lexer (part of the HTML lexer) sets the styling, but only has one word list (see [here](https://github.com/geany/geany/blob/32bbe106195f953e980bfa756b5a7ccad1ee209d/scintilla/lexilla/lexers/LexHTML.cxx#L724)) which is used as you noticed for the Php words.  The lexer is purely syntactic, it does not recognise semantics like function definitions, the ctags parser does that, but because there is only one wordlist available for php there is no way to pass the function list to the lexer to style those names as functions, so any setting in the theme will be unused.

**Comment by peterjosvai** [comment]
thank you so much for yor explanation...
  
(I am completely happy with the colors I have.. only would have styled functions if there was a way to do so...)
  

  
thanks, really
  

  
  

  
  
  
  
>
